File: php/features.php
<?php
declare(strict_types=1);

// IbanValidation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IbanValidationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IbanValidationBaseFeature();
            case "test":
                return new IbanValidationTestFeature();
            default:
                return new IbanValidationBaseFeature();
        }
    }
}

File: py/features.py
# IbanValidation SDK feature factory

from feature.base_feature import IbanValidationBaseFeature
from feature.test_feature import IbanValidationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IbanValidationBaseFeature(),
        "test": lambda: IbanValidationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
